master/internal/experiment.go

```diff
@@ -617,3 +608,23 @@ func (e *experiment) updateState(ctx *actor.Context, state model.State) bool {
 func (e *experiment) canTerminate(ctx *actor.Context) bool {
 	return model.StoppingStates[e.State] && len(ctx.Children()) == 0
 }
+
+func (e *experiment) Snapshot() (json.RawMessage, error) {
+	searcherSnapshot, err := e.searcher.Snapshot()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to snapshot searcher")
+	}
+	e.SearcherState = searcherSnapshot
+	experimentSnapshot, err := json.Marshal(e.experimentState)
+	return experimentSnapshot, errors.Wrap(err, "failed to marshal experiment")
+}
+
+func (e *experiment) Restore(experimentSnapshot json.RawMessage) error {
```

Comment:
I find this Snapshot vs snapshot and Restore vs restore naming to be very confusing.

Reply:
yeah. open to better names here, i'll try to come up with some.